backend/src/ee/services/access-approval-request/access-approval-request-service.ts

```diff
@@ -0,0 +1,360 @@
+
+    const approvers = await accessApprovalPolicyApproverDAL.find({
+      policyId: policy.id
+    });
+
+    const approverUsers = await userDAL.findUsersByProjectMembershipIds(
+      approvers.map((approver) => approver.approverId)
+    );
+
+    if (approverUsers.length !== approvers.length) {
+      throw new BadRequestError({ message: "Some approvers were not found" });
+    }
+
+    const duplicateRequest = await accessApprovalRequestDAL.findOne({
+      policyId: policy.id,
+      requestedBy: membership.id,
+      permissions: JSON.stringify(requestedPermissions),
+      isTemporary
+    });
+
+    if (duplicateRequest) {
+      if (duplicateRequest.privilegeId) {
+        const privilege = await additionalPrivilegeDAL.findById(duplicateRequest.privilegeId);
+
+        const isExpired = new Date() > new Date(privilege.temporaryAccessEndTime || ("" as string));
+
+        if (!isExpired || !privilege.isTemporary) {
+          throw new BadRequestError({ message: "You already have an active privilege with the same criteria" });
+        }
+      } else {
+        const reviewers = await accessApprovalRequestReviewerDAL.find({
+          requestId: duplicateRequest.id
+        });
+
+        const isRejected = reviewers.some((reviewer) => reviewer.status === ApprovalStatus.REJECTED);
+
+        if (!isRejected) {
+          throw new BadRequestError({ message: "You already have a pending access request with the same criteria" });
+        }
+      }
+    }
+
+    const approval = await accessApprovalRequestDAL.transaction(async (tx) => {
+      const approvalRequest = await accessApprovalRequestDAL.create(
+        {
+          policyId: policy.id,
+          requestedBy: membership.id,
+          temporaryRange: temporaryRange || null,
+          permissions: JSON.stringify(requestedPermissions),
+          isTemporary
+        },
+        tx
+      );
+
+      await smtpService.sendMail({
+        recipients: approverUsers.filter((approver) => approver.email).map((approver) => approver.email!),
+        subjectLine: "Access Approval Request",
+
+        substitutions: {
+          projectName: project.name,
+          requesterFullName: `${requestedByUser.firstName} ${requestedByUser.lastName}`,
+          requesterEmail: requestedByUser.email,
+          isTemporary,
+          ...(isTemporary && {
+            expiresIn: ms(ms(temporaryRange || ""), { long: true })
+          }),
+          secretPath,
+          environment: envSlug,
+          permissions: accessTypes,
+          approvalUrl: `${cfg.SITE_URL}/project/${project.id}/approval`
+        },
+        template: SmtpTemplates.AccessApprovalRequest
+      });
+
+      return approvalRequest;
+    });
+
+    return { request: approval };
+  };
+
+  const listApprovalRequests = async ({
+    projectSlug,
+    authorProjectMembershipId,
+    envSlug,
+    actor,
+    actorOrgId,
+    actorId,
+    actorAuthMethod
+  }: TListApprovalRequestsDTO) => {
```

Comment:
I think we should add authorization resource check to this endpoint function. I don't think everyone should be able to view the list of approval requests for a project.